runtime/ops/fs.rs

```diff
@@ -267,30 +267,37 @@ async fn op_write_file_async(
   data: ZeroCopyBuf,
   cancel_rid: Option<ResourceId>,
 ) -> Result<(), AnyError> {
-  let cancel_handle = match cancel_rid {
-    Some(cancel_rid) => state
-      .borrow_mut()
-      .resource_table
-      .get::<CancelHandle>(cancel_rid)
-      .ok(),
-    None => None,
-  };
   let (path, open_options) = open_helper(
     &mut state.borrow_mut(),
     &path,
     mode,
     Some(&write_open_options(create, append, create_new)),
     "Deno.writeFile()",
   )?;
+
   let write_future = tokio::task::spawn_blocking(move || {
     write_file(&path, open_options, mode, data)
   });
+
+  let cancel_handle = cancel_rid.and_then(|rid| {
```

Comment:
In https://github.com/denoland/deno/pull/17384 crowl added `ops.tryClose(rid)` calls after signalful read calls, which I presume might be doing the same that this is doing here.

I'm guessing this is probably the more performant of the two solutions. Just need to make sure that the "close handle after use" handling isn't duplicated.